### PR TITLE
Function in Javascript 

### LINKED ISSUE #3228: Functions in Javascript
### Description

Introduction to Functions in Javascript

### Domain

Frontend Dev HTML/CSS/JS

### Type of Contribution

Documentation

### Code of Conduct

- [X] I follow [Contributing Guidelines](https://github.com/girlscript/winter-of-contributing/blob/main/.github/CONTRIBUTING.md) & [Code of conduct](https://github.com/girlscript/winter-of-contributing/blob/main/.github/CODE_OF_CONDUCT.md) of this project.

### PR DESCRIPTION
**Pull request by geetika005**
<hr>

## Description 📜

Introduction to functions in javascript #3228  

<hr>

## Type of change 📝

- [ ] Au#dio (Should be in mp3 format Includes speech clarity, Concise ,Low distortion)
- [ ] Vi#deo (Animations, screen-recordings, presentations and regular explanatory films are all possibilties etc)
- [x] Documentation (Content Creation in the form of codes or tutorials)
- [ ] Other (If you choose other, Please mention changes below) 

<hr>

## Domain of Contribution 📊

- [ ] Android Dev #(Flutter)
- [ ] Android Dev #(Java)
- [ ] Android Dev #(Kotlin)
- [ ] Backend Dev #(Java)
- [ ] Backend Dev #(.NET)
- [ ] Backend Dev #(PHP)
- [ ] Backend Dev #(Python)
- [ ] C/#CPP
- [ ] Competitive #Programming
- [ ] Cyber #Security
- [ ] DS#A
- [ ] Data#base
- [ ] Datascience with #Python
- [ ] Datascience with #R
- [x] Frontend Dev HTML/CSS/JS
- [ ] Frontend Dev #React/Angular/Vue
- [ ] Go#lang
- [ ] Interview #Prep
- [ ] Java_#Domain
- [ ] Java#Script
- [ ] ME#RN
- [ ] Machine #Learning
- [ ] Open #Source
- [ ] Python_#Domain
- [ ] Ru#st
- [ ] Statis#tics
- [ ] UI/#UX

<hr>

## Checklist ✅

- [x] I follow [Contributing Guidelines](https://github.com/girlscript/winter-of-contributing/blob/main/.github/CONTRIBUTING.md) & [Code of conduct](https://github.com/girlscript/winter-of-contributing/blob/main/.github/CODE_OF_CONDUCT.md) of this project.
- [x] I have performed a self-review of my own code or work.
- [x] I have commented my code, particularly in hard-to-understand areas.
- [x] My changes generates no new warnings.
- [x] I'm GWOC'21 contributor

<hr>

## Screenshots / Gif (Optional) 📸

<hr>

### CODE REVIEW

**Comment by geetika005** [comment]
@praveenscience @code-reaper08  Please review my pr.

**Comment by geetika005** [comment]
@RuchikaSuryawanshi7  this is a pr request you can not assign it to yourself 
Unassign yourself please

**Comment by RuchikaSuryawanshi7** [comment]
@praveenscience  could you please review my pull request.

**Comment by geetika005** [comment]
> @praveenscience could you please review my pull request.



> @praveenscience could you please review my pull request.

Why are you posting it here ?
This pr belongs to me

**Comment by geetika005** [comment]
@praveenscience @code-reaper08 please review my pr it's been 4 days .

**Review by code-reaper08 — CHANGES_REQUESTED** [review]
Looks like this is already in the repository [here](https://github.com/girlscript/winter-of-contributing/tree/Frontend_Web_Development_HTML_CSS_JS/Web_Development/Javascript/1.Functions_And_Scope) @geetika005
And also you need to attribute the sources, I think you are inspired from https://www.hhutzler.de/blog/javascript-tackling-this-object-and-nested-functions-by-samples/

**Comment by geetika005** [comment]
@code-reaper08  sir my documentation is quite different from the one in scope and functions ,only the function definition ,syntax of function and parameter part is repeated. If i remove them then is it be okay to get merged.

**Comment by code-reaper08** [comment]
> @code-reaper08 sir my documentation is quite different from the one in scope and functions ,only the function definition ,syntax of function and parameter part is repeated. If i remove them then is it be okay to get merged.

Yeah it's quite different, but I would say, if you could make it more interesting, like a audio( Podcast ) contribution. It will be super good!

What you say @2024-SANDHYA?

**Comment by geetika005** [comment]
Can i add my documentation as an extention to scopes and function file @code-reaper08

**Comment by geetika005** [comment]
Can i add all the extra topics in that file

**Review by 2024-SANDHYA — CHANGES_REQUESTED** [review]
No need to write this :

image

Add the reference link Please & yes that can be done @code-reaper08

**Comment by geetika005** [comment]
Updated the file with reference link @code-reaper08 @2024-SANDHYA

File: Web_Development/Javascript/Javascript_Functions.md [hunk 1]
```diff
@@ -0,0 +1,118 @@
...
+```js
+function addSquares(s1, s2) {
+  function square(x) {
+    return x * x;
+  }
+  return square(s1) + square(s2);
+}
+a = addSquares(2, 5); // returns 29
+b = addSquares(3, 4); // returns 25
+```
```
Comment (code-reaper08) [inline Web_Development/Javascript/Javascript_Functions.md:117]: This part is totally rephrased from https://developer.mozilla.org/en-US/docs/Web/JavaScript/Guide/Functions

Please attribute it too

**Review by code-reaper08 — CHANGES_REQUESTED** [review]
I've left some reviews, please look into it :-)

**Comment by geetika005** [comment]
Updated the file with the changes sir @code-reaper08

**Review by code-reaper08 — APPROVED** [review]

**Pull request merged**